calendarsubscription/store.go

```diff
@@ -0,0 +1,189 @@
+		return nil, err
+	}
+	cs.ID = uuid.NewV4().String()
+	cs, err = cs.Normalize()
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = wrapTx(ctx, tx, b.create).ExecContext(ctx, cs.UserID, cs.ID, cs.Name, cs.Config, cs.ScheduleID, cs.Disabled)
+	if err != nil {
+		return nil, err
+	}
+	return cs, nil
+}
+func (b *Store) FindOneForUpdateTx(ctx context.Context, tx *sql.Tx, id string) (*CalendarSubscription, error) {
+	var userID = permission.UserID(ctx)
+	err := permission.LimitCheckAny(ctx, permission.Admin, permission.MatchUser(userID))
+	if err != nil {
+		return nil, err
+	}
+	err = validate.UUID("CalendarSubscriptionID", id)
+	if err != nil {
+		return nil, err
+	}
+
+	var cs CalendarSubscription
+
+	row := wrapTx(ctx, tx, b.findOneUpd).QueryRowContext(ctx, id)
+	err = row.Scan(&cs.ID, &cs.Name, &cs.UserID, &cs.Disabled, &cs.Config)
+	if err != nil {
+		return nil, err
+	}
+	return &cs, nil
+}
+
+// UpdateSubscriptionTx updates a calendar subscription with given information.
+func (b *Store) UpdateSubscriptionTx(ctx context.Context, tx *sql.Tx, cs *CalendarSubscription) error {
+	err := permission.LimitCheckAny(ctx, permission.Admin, permission.MatchUser(cs.UserID))
+	if err != nil {
+		return err
+	}
+
+	n, err := cs.Normalize()
+	if err != nil {
+		return err
+	}
+
+	update, err := b.FindOneForUpdateTx(ctx, tx, cs.ID)
+	if err != nil {
+		return err
+	}
+	if n.ScheduleID != update.ScheduleID {
+		return validation.NewFieldError("ScheduleID", "cannot update schedule id of calendar subscription")
+	}
+	if n.LastAccess != update.LastAccess {
+		return validation.NewFieldError("Last Access", "cannot update last access of calendar subscription")
+	}
+	if n.UserID != update.UserID {
+		return validation.NewFieldError("UserID", "cannot update owner of calendar subscription")
+	}
+
+	_, err = wrapTx(ctx, tx, b.update).ExecContext(ctx, cs.ID, cs.UserID, cs.Name, cs.Disabled, cs.Config)
+
+	return err
+}
+
+// FindAll returns all calendar subscriptions of a user.
+func (b *Store) FindAll(ctx context.Context) ([]CalendarSubscription, error) {
+	var userID = permission.UserID(ctx)
+	err := permission.LimitCheckAny(ctx, permission.Admin, permission.MatchUser(userID))
+	if err != nil {
+		return nil, err
+	}
+
+	rows, err := b.findAll.QueryContext(ctx, userID)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	calendarsubscriptions := []CalendarSubscription{}
+	for rows.Next() {
+		var cs CalendarSubscription
+		err = rows.Scan(&cs.ID, &cs.Name, &cs.UserID, &cs.LastAccess, &cs.Disabled, &cs.ScheduleID, &cs.Config)
+		if err != nil {
+			return nil, err
+		}
+		calendarsubscriptions = append(calendarsubscriptions, cs)
+	}
+
+	return calendarsubscriptions, nil
+}
+
+// DeleteSubscriptionsTx removes calendar subscriptions with the given ids.
+func (b *Store) DeleteSubscriptionsTx(ctx context.Context, tx *sql.Tx, ids []string) error {
+	err := permission.LimitCheckAny(ctx, permission.Admin, permission.MatchUser(permission.UserID(ctx)))
```

Comment:
we can use just `permission.User` as matching against the current user is redundant.